packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -314,16 +370,36 @@ function useFocusContainment(scopeRef: RefObject<Element[]>, contain: boolean) {
 
       let focusedElement = document.activeElement;
       let scope = scopeRef.current;
-      if (!isElementInScope(focusedElement, scope)) {
+      if (!scope || !isElementInScope(focusedElement, scope)) {
         return;
       }
 
-      let walker = getFocusableTreeWalker(getScopeRoot(scope), {tabbable: true}, scope);
+      let scopeRoot = getScopeRoot(scope);
+      if (!scopeRoot) {
+        return null;
+      }
+      let walker = getFocusableTreeWalker(scopeRoot, {tabbable: true}, scope);
+      if (!focusedElement) {
+        return null;
+      }
       walker.currentNode = focusedElement;
       let nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode()) as FocusableElement;
       if (!nextElement) {
-        walker.currentNode = e.shiftKey ? scope[scope.length - 1].nextElementSibling : scope[0].previousElementSibling;
-        nextElement = (e.shiftKey ? walker.previousNode() : walker.nextNode())  as FocusableElement;
+        if (e.shiftKey) {
+          let currNode = scope[scope.length - 1].nextElementSibling;
+          if (!currNode) {
+            return null;
+          }
+          walker.currentNode = currNode;
+          nextElement = walker.previousNode() as FocusableElement;
+        } else {
+          let currNode = scope[0].previousElementSibling;
+          if (!currNode) {
+            return null;
+          }
+          walker.currentNode = currNode;
+          nextElement = walker.nextNode() as FocusableElement;
+        }
       }
```

Comment:
The logic in this file is certainly a lot more complex when you account for null. My question is whether `null` is truly a valid value, or if it is something that should (as it did before) throw as an unexpected state. If so, you could use the `!` operator or perhaps make functions like `getScopeRoot` throw on null instead of possibly returning null and that might simplify code like this down the line. Another example: we know because we rendered the sentinels that previous/next element sibling will always be there so it is probably safe to assume that.

Reply:
That makes a lot of sense. I'll update it.

Reply:
file looks a lot better overall. any reason this one in particular couldn't be updated to basically be what it was before? we know previousElementSibling will be there for example.